polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisOverlappingTableTest.java

```diff
@@ -0,0 +1,304 @@
+
+  private static final CatalogWrapper defaultCatalog = new CatalogWrapper("default");
+  private static final CatalogWrapper laxCatalog = new CatalogWrapper("lax");
+  private static final CatalogWrapper strictCatalog = new CatalogWrapper("strict");
+
+  /** Used to define a parameterized test config */
+  protected record TestConfig(
+      DropwizardAppExtension<PolarisApplicationConfig> extension,
+      CatalogWrapper catalogWrapper,
+      Response.Status response) {
+    public String catalog() {
+      return catalogWrapper.catalog;
+    }
+
+    private String extensionName() {
+      return (extension
+              .getConfiguration()
+              .getConfigurationStore()
+              .getConfiguration(null, PolarisConfiguration.ALLOW_TABLE_LOCATION_OVERLAP))
+          ? "lax"
+          : "strict";
+    }
+
+    /** Extract the first component of the catalog name; e.g. `default` from `default_123_xyz` */
+    private String catalogShortName() {
+      StringBuilder result = new StringBuilder();
+      for (char c : catalog().toCharArray()) {
+        if (c == '_') {
+          break;
+        }
+        result.append(c);
+      }
+      return result.toString();
+    }
+
+    @Override
+    public String toString() {
+      return String.format(
+          "extension=%s, catalog=%s, status=%s",
+          extensionName(), catalogShortName(), response.toString());
+    }
+  }
+
+  /* Used to wrap a catalog name, so the TestConfig's final `catalog` field can be updated */
+  protected static class CatalogWrapper {
+    public String catalog;
+
+    public CatalogWrapper(String catalog) {
+      this.catalog = catalog;
+    }
+
+    @Override
+    public String toString() {
+      return catalog;
+    }
+  }
+
+  @BeforeEach
+  public void setup(PolarisConnectionExtension.PolarisToken adminToken) {
+    userToken = adminToken.token();
+    realm = PolarisConnectionExtension.getTestRealm(PolarisServiceImplIntegrationTest.class);
+    defaultCatalog.catalog = String.format("default_catalog_%s", UUID.randomUUID().toString());
+    laxCatalog.catalog = String.format("lax_catalog_%s", UUID.randomUUID().toString());
+    strictCatalog.catalog = String.format("strict_catalog_%s", UUID.randomUUID().toString());
+    List.of(BASE_EXT, LAX_EXT)
+        .forEach(
+            EXT -> {
+              List.of(defaultCatalog, laxCatalog, strictCatalog)
+                  .forEach(
+                      c -> {
+                        StorageConfigInfo config =
+                            FileStorageConfigInfo.builder()
+                                .setStorageType(StorageConfigInfo.StorageTypeEnum.FILE)
+                                .build();
+                        CatalogProperties.Builder propertiesBuilder =
+                            CatalogProperties.builder()
+                                .setDefaultBaseLocation(String.format("%s/%s", baseLocation, c));
+                        if (!c.equals(defaultCatalog)) {
```

Comment:
I would put this first, it's a bit hard to read and it's the most important part of this config. The rest of the code here is the same for all of the catalogs correct?

// Catalog specific conf
// Generic Conf
// Create catalog
// Create namespace

It would also be nice to flatten this a bit if possible. It's quite nested now which makes it pretty hard for me to read. I am old though and my eyes are bad so ymmv.

Reply:
Good call, let me move the `StorageConfigInfo` under this.

I also don't love the nesting, and some of this setup is not necessary (not every test will use every catalog and every extension). But I want to avoid overcomplicating this test, too. If you see an obvious fix let me know and I'm keen to refactor.

Reply:
I hate to suggest this because in my heart I like functional approaches, but I think they just look bad in Java.
So what you could do is convert to for (...) 

That would save us a lot of indentation because we have too keep adding a new level every time we add to our functional change thanks to our style guide.
```java
List.of(BASE_EXT, LAX_EXT)
  .forEach(
    EXT -> {
      List.of(defaultCatalog, laxCatalog, strictCatalog)
        .forEach(
          c -> {
            //6 Levels of indentation
```


```java
for (ext : List.of(...)) {
  for (catalog : List.of()) {
  // 3 levels of indentation
  }
}
```

```scala
for ( ext <- (...); catalog <-(...)) {
 // 1 level of indentation :(
}
```

Other things to make this more readable,

We could do static imports of PolarisConfiguration. This would require adding some "ignores" for the style guide I believe but I honestly don't think we lose anything by doing those static imports.

Long term we probably want to extract some of this out into a base test class maybe? So a user can just do something like

BaseTestClass.createCatalog(name, properties)

Reply:
The beauty of Scala is that you don't have to port the whole thing from Java 😇